fdbserver/storageserver.actor.cpp

```diff
@@ -4447,15 +4448,23 @@ ACTOR Future<Void> fetchKeys(StorageServer* data, AddingShard* shard) {
 		// Get the history
 		state int debug_getRangeRetries = 0;
 		state int debug_nextRetryToLog = 1;
+		state int lastErrorCode = error_code_success;
 
 		// FIXME: The client cache does not notice when servers are added to a team. To read from a local storage server
 		// we must refresh the cache manually.
 		data->cx->invalidateCache(keys);
 
 		loop {
 			state Transaction tr(data->cx);
-			fetchVersion = data->version.get();
-
+			// fetchVersion = data->version.get();
+			if (lastErrorCode == error_code_transaction_too_old) {
+				Version GRVVersion = wait(tr.getRawReadVersion());
+				fetchVersion = std::max(GRVVersion, data->version.get());
```

Comment:
Does this only work for split shards? Could you add an example in the comment?

Reply:
This works whenever the dest SS falls behind the src SS.